moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp

```diff
@@ -317,10 +346,20 @@ void TrajectoryVisualization::update(float wall_dt, float ros_dt)
       animating_path_ = true;
       displaying_trajectory_message_ = trajectory_message_to_display_;
       changedShowTrail();
+      trajectory_slider_panel_->update(trajectory_message_to_display_->getWayPointCount());
     }
-    else if (loop_display_property_->getBool() && displaying_trajectory_message_)
-    {  // do loop? -> start over too
-      animating_path_ = true;
+    else if (displaying_trajectory_message_)
+    {
+      if (trajectory_slider_panel_->isVisible() && !loop_display_property_->getBool() &&
+          trajectory_slider_panel_->getSliderPosition() == displaying_trajectory_message_->getWayPointCount() - 1)
+      {  // show the last waypoint if the slider is enabled
+        display_path_robot_->update(
+            displaying_trajectory_message_->getWayPointPtr(displaying_trajectory_message_->getWayPointCount() - 1));
```

Comment:
I don't think we can get a slider position of -1, so the update function will never be called with a zero-length trajectory. Please correct me if I'm wrong.